### PR TITLE
Implement gitlab service slack

### PR DESCRIPTION
**Pull request by jorcau**
Hi,

Here is a new proposal for Slack Integration implementation. It reuses work started on #65. I've talked with @hypnoglow about doing another PR so I guess we could close #65 for this one.

It includes documentation and acceptance tests:
```
➜ make testacc TEST=./gitlab TESTARGS='-run=TestAccGitlabServiceSlack_*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./gitlab -v -run=TestAccGitlabServiceSlack_* -timeout 120m
=== RUN   TestAccGitlabServiceSlack_basic
--- PASS: TestAccGitlabServiceSlack_basic (8.23s)
=== RUN   TestAccGitlabServiceSlack_import
--- PASS: TestAccGitlabServiceSlack_import (6.95s)
PASS
ok  	github.com/terraform-providers/terraform-provider-gitlab/gitlab	15.209s
```

Thanks,

### CODE REVIEW

**Comment by roidelapluie** [comment]
FYI now the Acc tests are run in travis too.

**Comment by roidelapluie** [comment]
Can you please add an ACC test with default values? Current tests are great but too explicit.

**Comment by jorcau** [comment]
> FYI now the Acc tests are run in travis too.

This is awesome!

I added a test with only required field set as you asked.

File: website/docs/r/service_slack.html.markdown [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+* `wiki_page_events` - (Optional) Enable notifications for wiki page events.
+
+* `wiki_page_channel` - (Optional) The name of the channel to receive wiki page events notifications.
+
+## Importing projects
+
+You can import a service_slack state using `terraform import <resource> <project>`.  The
+`project` is the id of a project:
+
+    terraform import gitlab_service_slack.awesome_project 1
```
Comment (roidelapluie) [inline website/docs/r/service_slack.html.markdown]: maybe use  `terraform import gitlab_service_slack.slack 1` would be less confusing?

File: website/docs/r/service_slack.html.markdown [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  visibility_level = "public"
+}
+
+resource "gitlab_service_slack" "slack" {
+  project                    = "${gitlab_project.awesome_project.id}"
+  webhook                    = "https://webhook.com"
+  username                   = "myuser"
+  push_events                = true
+  push_channel               = "push_chan"
+  issues_events              = true
```
Comment (roidelapluie) [inline website/docs/r/service_slack.html.markdown]: what happens if you set this and not issue_channel?

File: website/docs/r/service_slack.html.markdown (hunk 1 above)
Comment (roidelapluie) [reply website/docs/r/service_slack.html.markdown]: Also please use a code block (```)

Comment (jorcau) [reply website/docs/r/service_slack.html.markdown]: This was a mistake! Good catch, thanks!

File: website/docs/r/service_slack.html.markdown (hunk 2 above)
Comment (jorcau) [reply website/docs/r/service_slack.html.markdown]: This is valid and would work but you're right that it's not really clear, so I made the change.

File: gitlab/resource_gitlab_service_slack.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+	log.Printf("[DEBUG] delete gitlab slack service for project %s", project)
+
+	_, err := client.Services.DeleteSlackService(project)
+	return err
+}
+
+func resourceGitlabServiceSlackImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	d.Set("project", d.Id())
+
```
Comment (roidelapluie) [inline gitlab/resource_gitlab_service_slack.go]: shouldnt we fetch the service at this point from gitlab?

Comment (jorcau) [reply gitlab/resource_gitlab_service_slack.go]: Not sure what you mean here.

From what I understand, the `resourceGitlabServiceSlackImportState` function should just define the ID or necessary parameters so the READ function can then be called to actually fetch the resource's details.

**Comment by rsrchboy** [comment]
@roidelapluie Are there any blockers to merging this?

**Comment by jorcau** [comment]
Hi @roidelapluie, do you have time to review & merge this anytime soon? Thanks!

**Pull request merged**